# HG changeset patch
# User Rik <rik@octave.org>
# Date 1379349486 25200
#      Mon Sep 16 09:38:06 2013 -0700
# Node ID 15d592c82abc53c2d7615f9bf8183b4463621a5c
# Parent  aa380b3803152e6070f486994615852ab5dfdd5a
test: Add test for bug #40036 to axis.m

* scripts/plot/axis.m: Add test for bug #40036 to axis.

diff --git a/scripts/plot/axis.m b/scripts/plot/axis.m
--- a/scripts/plot/axis.m
+++ b/scripts/plot/axis.m
@@ -605,8 +605,21 @@ endfunction
 %!   a = logspace (-5, 1, 10);
 %!   loglog (a, -a);
 %!   axis tight;
 %!   assert (axis (), [1e-5, 10, -10, -1e-5]);
 %! unwind_protect_cleanup
 %!   close (hf);
 %! end_unwind_protect
 
+## Test 'axis tight' with differently oriented, differently numbered data vecs
+## Bug #40036.
+%!test
+%! hf = figure ("visible", "off");
+%! unwind_protect
+%!   Z = peaks (linspace (-3, 3, 49), linspace (-2, 2, 29));
+%!   surf (Z);
+%!   axis tight;
+%!   assert (axis (), [1 49 1 29 min(Z(:)) max(Z(:))]);
+%! unwind_protect_cleanup
+%!   close (hf);
+%! end_unwind_protect
+
